RG34XX-SP: Update boot partitions, which fixes RAM & analog sticks.

mkasick opened the pull request:
Some notes on this:

The firmware posted on win.anbernic.com is a private Google Drive link.  I've requested access, but who knows.  Allegedly it's the same as the stock SD firmware, of which I have imaged a pristine copy.

The stock `boot_package` has a couple of changes to the device tree, notably:
- The model reports as "sun50iw9" instead of "Anbernic RG34XX-SP".
- The GPIO controller label is "ANBERNIC-keys" instead of "Anbernic RG34XX-SP Controller".

@acmeplus do you repack the device tree, or are these "legitimate" changes by the vendor?

~~There appears to be minor changes to `boot0`.  I didn't attempt do disassemble it to figure them out specifically--I thought they might possibly be RAM related, but the changes have no obvious effect.~~  Edit: I monkeypatched the updated `boot0` into my existing SD card incorrectly.  After I rebuilt and applied a full image, the updated `boot0` _does_ result in the system seeing the full 2 GB of RAM.

The kernel update fixes the analog sticks, which previously weren't working.  This must be an input driver hack since there doesn't appear to be any relevant device-tree updates.  When I ran `evtest` on the old kernel it only reported three `ABS` axes which didn't change on stick movement, but with the updated kernel it has four axes which correctly change.  I also ran [PadTest](https://github.com/ShendoXT/padtest) (with the controller type set to DualShock) and all the inputs and rumble worked correctly.

~~`/proc/meminfo` is still reporting `MemTotal: 996480 kB` even with these changes.  I haven't actually booted stock yet on this to see if it sees all 2 GB of RAM.  I'll give that a try sometime.~~  Edit: `/proc/meminfo` reports `MemTotal: 2027136 kB`.

@acmeplus What's your preferred way to handle the boot partitions?  The `boot0` and kernel update are helpful, but I'm not sure if you prefer the changes to `boot_package.fex`.

mkasick committed:
h700: rg34xx-sp: Update boot partitions.

Update boot0.img, boot_package.fex (u-boot & monitor), and
boot.img-zImage from stock (RG34XXSP-V1.0.1-EN64GB-250408).

Fixes detection of 2 GB RAM and analog sticks.

acmeplus commented:
@mkasick Yeah, I was not able to download a copy from their site since it's in a private drive, and as I mentioned I don't have a device yet, so I've been relying on users to do some testing.

About the boot_package, the changes to the DTS regarding the model and gpio controller are mine, I do those to keep a single controller map for each device (needed specifically for devices that lack analogues so we can define the equivalent analogue joystick when the dpad to analogue toggle is used). 

I didn't have a full copy of the sdcard firmware, only a portion of the boot loader and a copy of the kernel, so some bits were done based on the rg34xx firmware. If your changes work to detect 2GB, get analogue controls, and support for the lid I'm ok merging the partitions.

The only thing that I would revert is the controller map that you committed before, since it looks like you have removed that change in this PR.

Let me know if you are ok with those and then I'll merge once you update the PR.

Thanks!

mkasick commented:
@acmeplus I updated `boot_package.fex` with your original device-tree blob and removed the `es_input.cfg` change.  I did keep the updated `u-boot` and `monitor` builds since they come from the same stock source as `boot0` and the kernel, although I don't think the changes there are material.

I couldn't find any tool to parse a packed `boot_package.fex` so I [wrote one](https://gist.github.com/mkasick/b08306615dcabab87cf8033cce4783b8), if that's helpful to anyone else.

I'm not sure what's going on with the lid.  The hall-effect sensor does work, but the device doesn't automatically resume on lid open and requires pressing the power button.  The biggest issue there is that it stays asleep instead of powering off when unplugging the charger (while not booted).

acmeplus commented:
@mkasick Thanks for the update, I'll merge the PR. I won't be able to investigate until I get my unit, but @Mikhailzrick may have some ideas about the lid since he worked on those changes in the SP.

Thanks for the boot-package.fex extraction tool I was just using a simple rudimentary script with manually extracted offsets, so that's much better.

Mikhailzrick commented:
My guess is the hall sensor is not a wakeup event, or it’s not enabled. There was a time the 35xx-sp was the same. We’ve not done anything for it one way or the other, it’s just how it behaves from anbernic depending on I’m assuming the kernel version used.
